rewrite-xml/src/main/java/org/openrewrite/xml/security/DateBoundSuppressions.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.xml.security;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.marker.Markers;
+import org.openrewrite.xml.XmlIsoVisitor;
+import org.openrewrite.xml.tree.Xml;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class DateBoundSuppressions extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Add date bounds to OWASP suppressions";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Adds an expiration date to all OWASP suppressions in order to ensure that they are periodically reviewed. " +
+                "More details: https://jeremylong.github.io/DependencyCheck/general/suppression.html";
+    }
+
+    @Option(displayName = "Until date",
+            description = "Optional. The date to add to the suppression. Default will be 30 days from today.",
+            example = "2023-01-01")
+    @Nullable
+    String untilDate;
+
```

Comment:
Do these suppression files have a particular name or location within a repository?

Feels like there should be an applicability test preventing this from modifying/wasting time traversing totally unrelated XML files that happen to have a tag named "suppress".

Reply:
Good catch, has to be at the root of the repo, and has to have the xsd `https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.3.xsd` (although i'm thinking to regex match so it isn't required to have 1.3). Will add an applicability test (is there a `suppressions.xml` at the root of the repo?) and a single source applicability test (is this the `suppressions.xml` and does it have `xmlns=https://jeremylong.github.io/DependencyCheck/dependency-suppression.(.*?).xsd`?).

Reply:
Added in latest commit, thanks for the suggestion!